src/EnergyPlus/FanCoilUnits.cc

```diff
@@ -4942,6 +5015,55 @@ namespace FanCoilUnits {
         return Residuum;
     }
 
+    Real64 CalcFanCoilHeatCoilPLRResidual(EnergyPlusData &state, Real64 const CyclingR,  // electric heating coil cycling ratio
+        Array1D<Real64> const &Par // Function parameters
+    )
+    {
+        // PURPOSE OF THIS SUBROUTINE:
+        // Calculate electric heating coil cycling ratio of FanCoilUnit with MultiSpeedFan 
+        // capacity control method when running with at lowest speed for a continuous fan 
+        // fan operating mode.
+
+        // METHODOLOGY EMPLOYED:
+        // Use SolveRoot to CALL this Function to converge on a solution
+
+        // Return value
+        Real64 Residuum; // Result (force to 0)
+
+        // SUBROUTINE LOCAL VARIABLE DECLARATIONS:
+        int FanCoilNum;          // Index to this fan coil unit
+        bool FirstHVACIteration; // FirstHVACIteration flag
+        int ZoneNum;             // controoled zone index
+        int WaterControlNode;    // water node to control
+        Real64 QZnReq;           // Sensible load to be met [W]
+        Real64 QUnitOut;         // delivered capacity [W]
+        Real64 PLR;              // fan coil unit PLR
+
+        // convert parameters to usable variables
+        FanCoilNum = int(Par(1));
+        if (Par(2) == 1.0) {
+            FirstHVACIteration = true;
+        } else {
+            FirstHVACIteration = false;
+        }
+        ZoneNum = int(Par(3));
+        QZnReq = Par(4);
+        WaterControlNode = int(Par(5));
+        PLR = 1.0;
+
+        // electric heating coil cycling ratio at minimum air flow for constant fan operating mode 
+        Calc4PipeFanCoil(state, FanCoilNum, ZoneNum, FirstHVACIteration, QUnitOut, PLR, CyclingR);
+
+        // Calculate residual based on output magnitude
+        if (std::abs(QZnReq) <= 100.0) {
+            Residuum = (QUnitOut - QZnReq) / 100.0;
+        } else {
+            Residuum = (QUnitOut - QZnReq) / QZnReq;
+        }
+
+        return Residuum;
+    }
+
```

Comment:
This new function was added too calculate the heating load residual and determine the *cycling ratio* using the root solver.